### DISCUSSION TITLE
Executing npm start with Remote development gives broken URL

### DISCUSSION THREAD

**Discussion by BillyCottrell** [loopbackio/loopback-next#7462] [Q&A] [answered]
Hi there,

So I am trying to do remote development on my Linux server using VS Code remote SSH plugin. Now when do npm start with my Angular application I get the normal localhost URL, but when I try to do the same with my Loopback it gives the following URL http://[::1]:3000 which doesn't work. What would you advise in this situation, should I stop using the npm start command and let it run using pm2 ecosystem instead or something else?

Kind regards,

Billy Cottrell

**Comment by raymondfeng** [comment, marked as answer]
You can use HOST env var to configure the server hostname/ip.

**Reply by BillyCottrell** [reply]
Hi @raymondfeng,

So I did as you mentioned I was able to get it to work for development (actually its only locally so a different pc on the same network can't connect to it). In production though I still haven't figured out how to set this up. So I started using pm2 but still unable to get it to work. The thing is it's a local server which will be used as a LAN server so domain names aren't used to connect with the server I just use it's ip address. Normally you would have an API on an URL like api.example.com., but in this case its rather 1.1.1.1/api-example and when I try to use it this way I get the following:

> NotFoundError
> 404 Endpoint "GET /test-server-manager-api" not found.

I thought that If I would configure my nginx config file with the path included that it wouldn't be an issue, which isn't the case:
```
server {
	listen 80;
	listen [::]:80;
	root /var/www/html;
	index index.html index.htm index.nginx-debian.html;

	server_name _;
	location / {
		try_files $uri $uri/ = 404;
	}
	# regular webapp in production works
	location /server-manager-app{
		alias /var/www/server-manager-app;
		try_files $uri $uri/ /index.html;
	}
	# This is the part that doesn't work:
	location /server-manager-api {
		alias /var/www/server-manager-api;
		proxy_pass http://127.0.0.1:3000;
	}
}
```

Any ideas why this won't work? I tried different combinations of HOST values even looked if I needed to change the base path, but I have no clue what I should change to achieve the same situation.

Thanks in advance,

Billy Cottrell

**Reply by achrinza** [reply]
Apologies for the short "look over there" response, but perhaps this StackExchange answer may help:

https://serverfault.com/a/870620

**Reply by BillyCottrell** [reply]
No problem, cause this actually solved the issue thanks a lot!

**Comment by raymondfeng** [comment]
Is your Nginx running on the server as the LoopBack application? If not, `127.0.0.1` won't work and you need to specify an ip address or hostname that allows Nginx to connect to the LoopBack application. The ip address can be private between Nginx and LoopBack.

**Reply by BillyCottrell** [reply]
Both Nginx and loopback is running on the same server which is why I thought to use that ip address. Nonetheless, it doesn't work, so I thought to use a different ip address but failed as well. pm2 gave me an error message telling me that such address isn't available. So I have no clue what I have to do to achieve this.
